.github/workflows/README.md

```diff
@@ -0,0 +1,115 @@
+# Workflow Taxonomy
+
+## Where and what tests are run
+
+---
+
+### Workflow: `deploy-happy-stack.yml`
+
+#### Ran on:
+
+- `deployment`: (deployments are triggered by `push-tests`)
+
+#### Jobs:
+
+- ##### `upgrade`: TODO
+- ##### `e2e-test`:
+  - Runs playwright against deployed frontend app uploading any artifacts
+    - ```bash
+            if [ "${DEPLOYMENT_STAGE}" == "stage" ]; then
+            export DEPLOYMENT_STAGE=staging
+          fi
+          if [ "${DEPLOYMENT_STAGE}" != "prod" ]; then
+            pip3 install -r scripts/smoke_tests/requirements.txt
+            python3 -m scripts.smoke_tests.setup
+          fi
+          cd frontend
+          npm ci
+          npx playwright install --with-deps
+          cp src/configs/${DEPLOYMENT_STAGE}.js src/configs/configs.js
```

Comment:
In what case is this config necessary?

Reply:
src/configs/configs.js defines which backend API base URL to use for any backend calls (local vs. a deployed env) + a set-up config for plausible analytics

Reply:
but the frontend is never started, so there's no reading of the config? @nayib-jose-gloria @tihuan

Reply:
I believe its playwright uses it as well for certain tests, at least the API_URL

Reply:
we can probably refactor that so it isn't the case, but as-is we'd need to change a few things

Reply:
Ah test URL is configured through `TEST_ENV`, so it doesn't use `API_URL` I think!

```ts
const TEST_ENV_TO_TEST_URL = {
  dev: "https://cellxgene.dev.single-cell.czi.technology",
  happy: "http://frontend.corporanet.local:3000",
  local: "http://localhost:3000",
  localProd: "http://localhost:9000",
  prod: "https://cellxgene.cziscience.com",
  rdev: process.env.RDEV_LINK || "",
  staging: "https://cellxgene.staging.single-cell.czi.technology",
};
```

Reply:
here's one example: https://github.com/chanzuckerberg/single-cell-data-portal/blob/d5c6377d407957db32e82bf5152cea943680d02b/frontend/tests/features/collection/collection.test.ts#L4

Reply:
those TEST_ENV_TO_TEST_URL values also aren't the base URLs you'd use to wait for a response from a particular backend API endpoint, which is how we fixed the flakey invalid DOI tests

Reply:
Ahh I see! Sorry totally missed that!

What if we remove `API_URL` import in tests and update the assertion to be:

```ts
const collectionEndpointPath = `dp/v1/collections`; // <-- removed API_URL

/**
 * Submit create invalid collection form.
 * @returns Form submit invalid parameters error response (400).
 */
async function submitCreateFormInvalid(page: Page) {
  return await Promise.all([
    page.waitForResponse(
      (response) =>
        response.url().includes(collectionEndpointPath) &&  // <-- update to match partial path
        response.status() === HTTP_STATUS_CODE.BAD_REQUEST
    ),
    page.click(getTestID("create-button")),
  ]);
}
```

Reply:
If that looks good I can add the change in this PR!

https://github.com/chanzuckerberg/single-cell-data-portal/pull/3516

Reply:
yeah, that looks good to me!

Reply:
Awesome! Since that PR is now merged, I'll create a separate PR to address that today 👍 Thank you!

Reply:
PR out! https://github.com/chanzuckerberg/single-cell-data-portal/pull/3531 PTAL thank you 🙏